python/cudf/cudf/core/dtypes.py

```diff
@@ -268,6 +268,10 @@ def __init__(self, precision, scale=0):
         self._validate(precision, scale)
         self._typ = pa.decimal128(precision, scale)
 
+    @property
+    def str(self):
+        return np.dtype("int64").str
```

Comment:
This feels off to me. Do we need to maintain compatibility with `__array_interface__` for this property?

Reply:
Honestly I am not 100% sure what to put for this. There's not a `typestr` that corresponds to decimal. In lieu of that, since the underlying data is `int64`, it seemed like the only option that made any sense. At least then then consuming interface knows what chunk of memory is relevant, even if it has to combine it with some metadata to view the values as they are intended separately.

Reply:
In pandas there's a (largely unused?) decimal type and they just put `decimal` for this

https://github.com/pandas-dev/pandas/blob/b7cb3dc25a5439995d2915171c7d5172836c634e/pandas/tests/extension/decimal/array.py#L39

Not sure how it'd be more useful to do that though.

Reply:
What if we did:
```
@property
def str(self):
    return f"decimal64({self.precision}, {self.scale})"
```

It generally matches PyArrow then, and nothing handles `"decimal"` anyway.